assets/js/hooks/youtube_player.js

```diff
@@ -0,0 +1,139 @@
+/**
+ * Contains client-side logic for the youtube iframe embeded player.
+ * It shall contain multiple hooks, all revolving around various youtube player
+ * functionalities that we wish to have.
+ */
+export const RenderYouTubePlayer = {
+  mounted() {
+    console.log(">>> mounted YouTubePlayer JS-Hook!!", this);
+    // 2. This code loads the IFrame Player API code asynchronously.
+    let tag = document.createElement("script");
+    const foo = document.getElementById("player");
+    console.log("FOO before: ", foo);
+    // tag.crossOrigin = 'anonymous'; /// seems like this will prevent the CORS allow origin from youtube to work correctly since it does the opposite (by allowing origin-only) REF: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/crossorigin
+    tag.src = "https://www.youtube.com/iframe_api";
+    console.log(">>> script tags:", document.getElementsByTagName("script"));
+    /// ensures that the API script is loaded before any subsequent scripts that depend on it, hence we insert before the first script tag:
+    let firstScriptTag = document.getElementsByTagName("script")[0];
+    firstScriptTag.parentNode.insertBefore(tag, firstScriptTag);
+
+    console.log(
+      ">> after insertion, script tags:",
+      document.getElementsByTagName("script"),
+    );
+    let iframeInitialiserScript = document.createElement("script");
+    document.body.appendChild(iframeInitialiserScript);
+    const stringifiedScript = `
```

Comment:
solid stuff, that's much cleaner 

point to note: because the code in the script tag gets evaluated elsewhere, it won't have a reference in the same scope to the functions defined prior. Hence, I've kept the callbacks under `window.*` to circumvent that.